mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/binding/impl/AbstractBindingContext.java

```diff
@@ -34,23 +34,14 @@ public BeanResolver beanResolver() {
 	}
 
 	@Override
-	public <T> T param(String name, Class<T> paramType) {
-		Contracts.assertNotNull( name, "name" );
-		Contracts.assertNotNull( paramType, "paramType" );
-
-		Object value = params.get( name );
-		if ( value == null ) {
-			throw log.paramNotDefined( name );
-		}
-
-		return paramType.cast( value );
+	public NamedValues params() {
+		return params;
 	}
 
-	@Override
-	public <T> Optional<T> paramOptional(String name, Class<T> paramType) {
-		Contracts.assertNotNull( name, "name" );
-		Contracts.assertNotNull( paramType, "paramType" );
+	private static class BindingContextParameters extends MapNamedValues {
 
-		return Optional.ofNullable( params.get( name ) ).map( paramType::cast );
+		public BindingContextParameters(Map<String, Object> values) {
+			super( values, log::paramNotDefined );
+		}
```

Comment:
This could have been a factory method!

(my version of "this could have been an email")